### ISSUE TITLE
Self-pin topics

### ISSUE THREAD

**Issue by mybearworld** [ScratchAddons/ScratchAddons#6497]
### Why this would be helpful

Sometimes, I want to keep up with certain topics even if they're not on the first page, but not necessarily follow them as that clogs up my messages. 

### How this addon works

This addon would allow you to pin any post to your own personal "Pinned" section in every forum category (e.g. pins in Suggestions and BaG would stay separate, for example)

**Comment by DNin01** [comment]
Browser bookmarks can also help, but having a section of pinned topics would allow you to see new activity at a glance.

**Comment by mxmou** [comment]
Duplicate of #5353

**Comment by WorldLanguages** [comment]
Scratch Forons

**Comment by mybearworld** [comment]
> Scratch Forons

?

**Comment by DNin01** [comment]
> Scratch Forons

Forum addons?

**Comment by WorldLanguages** [comment]
Dunno, maybe this shouldn't be rejected, we could build an intuitive UI over this. But it still sounds painful to make it work. And it's probably blocked by the absence of addon.storage so that pins sync through devices.

**Comment by DNin01** [comment]
Regarding staying below synced storage limits, at a minimum, we'd probably only have to store an array of topic IDs. Each one is currently up to 6 digits long, so if I did my math right, we can store 1,169 of them in a single storage key. Once there are over a million topics, they'll become 7 digits long. In that case, 1,023 topics could be stored, so probably we could add a limit of 1,000 topics.

**Comment by DNin01** [comment]
I think this feature would be useful mainly for keeping track of new activity, since ideally the list would have dots indicating new posts.

But the way I do it right now is I can take a link to the topics I've posted on or the topics I'm following. That kind of covers it for me and all I have to do to save a topic is to follow it.

**Comment by BroJac5246** [comment]
So is this still rejected? Or is it not rejected but requires serious consideration before being added? If it's the latter then can this be reopened?

**Comment by BroJac5246** [comment]
> This suggestion is still a shame though, as one would think that advanced forum users would know about other extensions and browser features to save a webpage for reading later. Or maybe those aren't good enough.
> My point is, this suggestion could really apply to everything: watching a project later, looking at a studio later, and so on... So one would think we would have something better than bookmarks that comes to mind, and we do not. I guess browser vendors should try harder.

There are benefits to this occurring directly on Scratch, such as being able to see if the topic has new posts.
